tock/hours/float.py

```diff
@@ -0,0 +1,147 @@
+import datetime as dt
+
+from .models import ReportingPeriod
+
+from employees.models import UserData
+from tock.utils import get_float_data, flatten
+
+# Variables.
+BASE_HOURS_PER_WEEK = 32.5
+BASE_HOURS_PER_DAY = 6.5
+BASE_DAYS_PER_WEEK = 5
+
+def float_tasks_for_view(request, kwargs):
+    rp = ReportingPeriod.objects.get(start_date=kwargs['reporting_period'])
+    userdata = UserData.objects.get(user=request.user)
+    float_people_id = get_float_people_id(userdata)
+    if float_people_id:
+        float_tasks = get_float_tasks(
+            start_date=rp.start_date,
+            float_people_id=float_people_id,
+            weeks=1
+        )
+        float_holidays = get_float_holidays(
+            start_date=rp.start_date,
+            weeks=2
+        )
+        float_timeoffs = get_float_timeoffs(
+            end_date=rp.end_date,
+            float_people_id=float_people_id,
+            weeks=26
+        )
+        work_days = get_work_days(
+            holidays=float_holidays,
+            start_date=rp.start_date,
+            end_date=rp.end_date
+        )
+        work_hours = get_work_hours(
+            start_date=rp.start_date,
+            end_date=rp.end_date,
+            timeoffs=float_timeoffs,
+            work_days=work_days
+        )
+        return get_tasks(
+            start_date=rp.start_date,
+            end_date=rp.end_date,
+            tasks=float_tasks,
+            work_days=work_days,
+            work_hours=work_hours
+        )
+    return None
```

Comment:
stylistic pref (though i also think it's easier to parse): move the `return None` failsafe above the code that does the meaty stuff:

```
if float_people_id:
    return None

float_tasks = ...
return something_awesome
```